### DISCUSSION TITLE
Student project problem

### DISCUSSION THREAD

**Discussion by Jlester92** [TEALSK12/introduction-to-computer-science#432] [General]
Student has a script that is to reset the game.  For some reason the answer block isn't storing his answer.  Any ideas?

(https://snap.berkeley.edu/snap/snap.html#present:Username=akaputz23&ProjectName=hggf)

**Comment by jpfasano** [comment]
This student should give meaningful names to all their sprites.  If this had been done, it would be easier to explain the issue.

The problem is that the sprite named Sprite has a forever loop that is continuously broadcasting 'game over'.
The sprite named 'start/tutorul page' receives the 'game over' message many times (10s if not 100s of times).
Each time 'start/tutorul page' receives the 'game over' message the "ask _want to restart?_ and wait" block is run. 
Running this ask block many times with the same question is the reason the student believes "the answer block isn't storing his answer".

A fix is to change "broadcast 'game over'" to be "broadcast 'game over' and wait".
![Screenshot 2022-12-13 132719](https://user-images.githubusercontent.com/9619225/207415300-26ebf2f5-8f91-4554-9f77-e1cfe50d3979.png)

And adding "stop all but this script" here:
![Screenshot 2022-12-13 132948](https://user-images.githubusercontent.com/9619225/207415830-191dd2a9-6e52-4d23-9205-e209c161e51b.png)
